File: data/evaluation/geeks_for_geeks_successful_test_scripts/cpp/FIND_POSITION_GIVEN_NUMBER_AMONG_NUMBERS_MADE_4_7.cpp
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#include <iostream>
#include <cstdlib>
#include <string>
#include <vector>
#include <fstream>
#include <iomanip>
#include <bits/stdc++.h>
using namespace std;
int f_gold ( string n ) {
  int i = 0, pos = 0;
  while ( n [ i ] != '\0' ) {
    switch ( n [ i ] ) {
      case '4' : pos = pos * 2 + 1;
      break;
      case '7' : pos = pos * 2 + 2;
      break;
    }
    i ++;
  }
  return pos;
}


//TOFILL

int main() {
    int n_success = 0;
    vector<string> param0 {"7","305745689","444","4","2074","27","447","255","10000111111011","fAKcSDRTNz"};
    for(int i = 0; i < param0.size(); ++i)
    {
        if(f_filled(param0[i]) == f_gold(param0[i]))
        {
            n_success+=1;
        }
    }
    cout << "#Results:" << " " << n_success << ", " << param0.size();
    return 0;
}

File: data/evaluation/geeks_for_geeks_successful_test_scripts/python/FIND_POSITION_GIVEN_NUMBER_AMONG_NUMBERS_MADE_4_7.py
# Copyright (c) 2019-present, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the license found in the
# LICENSE file in the root directory of this source tree.
#
def f_gold ( n ) :
    i = 0
    j = len ( n )
    pos = 0
    while ( i < j ) :
        if ( n [ i ] == '4' ) :
            pos = pos * 2 + 1
        if ( n [ i ] == '7' ) :
            pos = pos * 2 + 2
        i = i + 1
    return pos


#TOFILL

if __name__ == '__main__':
    param = [
    ('7',),
    ('305745689',),
    ('444',),
    ('4',),
    ('2074',),
    ('27',),
    ('447',),
    ('255',),
    ('10000111111011',),
    ('fAKcSDRTNz',)
        ]
    n_success = 0
    for i, parameters_set in enumerate(param):
        if f_filled(*parameters_set) == f_gold(*parameters_set):
            n_success+=1
    print("#Results: %i, %i" % (n_success, len(param)))